src/Dapr.Workflow/WorkflowServiceCollectionExtensions.cs

```diff
@@ -126,6 +152,40 @@ static bool TryGetGrpcAddress(out string address)
             address = string.Empty;
             return false;
         }
+
+        static GrpcChannel CreateChannel(HttpClient client)
+        {
+        string address = "localhost";
+        string? daprPortStr = Environment.GetEnvironmentVariable("DAPR_GRPC_PORT");
+        if (int.TryParse(daprPortStr, out int daprGrpcPort))
+        {
+            // There is a bug in the Durable Task SDK that requires us to change the format of the address
+            // depending on the version of .NET that we're targeting. For now, we work around this manually.
+#if NET6_0_OR_GREATER
+            address = $"http://localhost:{daprGrpcPort}";
+#else
+            address = $"localhost:{daprGrpcPort}";
+#endif
+        }
+            GrpcChannelOptions options = new() { HttpClient = client};
+            return GrpcChannel.ForAddress(address, options);
+        }
+    }
+
+    static class DaprDurableGrpcExtensions
+    {
+        const string ClientName = "durabletask-grpc";
+
+        public static IServiceCollection ConfigureDurableGrpcClient(this IServiceCollection services, string apiToken)
+        {
+            services.AddHttpClient(ClientName)
```

Comment:
For my understanding: Why do we need to add HttpClient while configuring Grpc Client?

Reply:
Under the hood, the gRPC client uses the HttpClient. It's all HTTP/2 for transport iirc.